### ISSUE TITLE
Pathfinding with Recast and Detour [Crowd/Groups]

### ISSUE THREAD

**Issue by gpascualg** [eloipuertas/ES2015A#238]
### Description

Moving more than one agent at once might come with implicit problems, such as collitions between them, all of them trying to reach the same point, not reorganizing to fit in a narrow path...

Crowd(s) attempt to fix that by implementing an steering behaviour on all agents in the group. It does come with certain limitations:
- Once in a crowd, the agent no longer has any control over its movements, the crowd completely takes control of it.

I don't believe this to be a problem, as our approach already gives Unity's navmesh all the control.

The following crowds structure is propossed:
- Player will have a big `DetourCrowd`, where every single unit will be added to
- AI will have a big `DetourCrowd`, where every single unit will be added to
- Both will:
  - When moving a single unit, the agent will be moved particularly, aside from the group (though still being _in_ it)
  - When moving a group of agents, all those agents will move as a crowd, but they'll still belong to the same big crowd.

Basically, to avoid having crowds in crowds (ie. selecting a subgroup of a group forming a new crowd), selections will be handled by taking slices of that bug group.
### Acceptance Criteria
- [x] An agent can be added to group, and removed if necessary
- [x] An agent can be fetched of a group
- [x] Moving a single/group of agents without moving the entire group should be possible

**Estimated time effort: 4h**

**Comment by gpascualg** [comment]
Bones a tots!

Primer de tot informar que el pathfinding ja està funcionant. Ara per ara compila però cal canviar certs components de tots els prefabs. S'haurà de fer quan es faci el merge a devel, per tant seria interesant que el dia que ho fem, espero que com a tard demà, estiguem uns quants de cada grup, per possibles problemes que puguin sortir, etc.

La qüestió es, caldrà:
- Canviar tots els `NavMeshBlock` per `Pathfinding > DetourBlock`, i, és clar, ajustar les dimensions. El `DetourBlock` ha de ser més gran que el `Collider`, aproximadament ha de tenir un marge un pel superior a les dimensions del `Collider` de les unitats. Sinó, els colliders es toquen i passen coses rares (ie. edificis volant :+1:)
- Canviar tots els `NavMeshAgent` per `DetourAgent`. Per ara es poden deixar els valors per defecte que hi venen posats.

Com dic, això s'haurà de fer un cop estigui el merge fet. Per ara, solament informo.

Sort a tots!

**Comment by gpascualg** [comment]
L'algorisme de control de grups, `DetourCrowd`, pren control total del moviment de les unitats, el que inclou posició i velocitat. Per tant, en endavant, caldrà marcar també la casella de constraints per l'`axis Y` i que ignori la gravetat, tot plegat en el component `RigidBody`.

Caldrà fer-ho a partir de quan es faci l'integració a devel, amb sort demà, tingueu-ho en compte per totes les unitats!
